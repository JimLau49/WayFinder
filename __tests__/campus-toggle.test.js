import React from "react";
import CampusToggle from "../src/components/campus-toggle/campus-toggle.component";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("CampusToggle component", () => {
  it("CampusToggle: snapshot renders correctly", () => {
    const tree = renderer.create(<CampusToggle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("onSGWPressButton: isSGW state hook is set to true", () => {
    // Allows to render a react component as an object in memory instead of the DOM (makes it faster)
    // wrapper allows us to easily examine the rendered components
    const mockFn = jest.fn();
    const wrapper = shallow(<CampusToggle onCampusToggle={mockFn} />);
    const button = wrapper.find("View TouchableHighlight").at(0);
    button.simulate("press");
    expect(mockFn).toHaveBeenCalledTimes(1);

    // Since we cannot directly check the state of react hooks, we test side effects instead
    // In this case, the styling changes of the buttons are the side effect
    const viewSGWStyling = wrapper
      .find("View TouchableHighlight View")
      .at(0)
      .props();
    expect(viewSGWStyling).toHaveProperty(
      ["style", 0, "backgroundColor"],
      "#AA2B45"
    );
    const ViewLoyolaStyling = wrapper
      .find("View TouchableHighlight View")
      .at(1)
      .props();
    expect(ViewLoyolaStyling).toHaveProperty(
      ["style", 0, "backgroundColor"],
      "#F2F2F2"
    );
  });

  it("onLoyolaPressButton: isSGW state hook is set to false", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<CampusToggle onCampusToggle={mockFn} />);
    const button = wrapper.find("View TouchableHighlight").at(1);
    button.simulate("press");
    expect(mockFn).toHaveBeenCalledTimes(1);

    // Since we cannot directly check the state of react hooks, we test side effects instead
    // In this case, the styling changes of the buttons are the side effect
    const viewSGWStyling = wrapper
      .find("View TouchableHighlight View")
      .at(0)
      .props();
    expect(viewSGWStyling).toHaveProperty(
      ["style", 0, "backgroundColor"],
      "#F2F2F2"
    );
    const ViewLoyolaStyling = wrapper
      .find("View TouchableHighlight View")
      .at(1)
      .props();
    expect(ViewLoyolaStyling).toHaveProperty(
      ["style", 0, "backgroundColor"],
      "#AA2B45"
    );
  });
});
