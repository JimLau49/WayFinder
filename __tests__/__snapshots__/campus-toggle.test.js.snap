// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampusToggle component should match snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 80,
      "position": "absolute",
      "width": 750,
      "zIndex": 0,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID="SGWButton"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#AA2B45",
            "flex": 1,
            "width": 375,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#F2F2F2",
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingTop": 25,
              "textAlign": "center",
            },
          ]
        }
      >
        SGW
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID="loyolaButton"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "flex": 1,
            "width": 375,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#AA2B45",
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingTop": 25,
              "textAlign": "center",
            },
          ]
        }
      >
        Loyola
      </Text>
    </View>
  </View>
</View>
`;
